### ISSUE TITLE
rlImGui doesn't respect WantCaptureMouse/WantCaptureKeyboard

### ISSUE THREAD

**Issue by albertvaka** [raylib-extras/rlImGui#63]
When you click inside an ImGui window, that click should be "consumed" by ImGUI and not delivered to the app (ie: `IsMouseButtonPressed()` should return `false` after that).

ImGUI uses the flag `WantCaptureMouse` to signal it wants to capture mouse movement and clicks, but rlImGui doesn't currently read this flag. This is explained in [ImGUI's FAQ](https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-how-can-i-tell-whether-to-dispatch-mousekeyboard-to-dear-imgui-or-my-application).

Same goes for keystrokes and the flag `WantCaptureKeyboard`.

Note: This doesn't seem possible to implement on top of the current Raylib API, new API methods might be needed.

**Comment by JeffM2501** [comment]
This is no how ImGui works. you are supposed to check ImGui if it wants the device input or not. The IO has want key and want mouse fields. If ImGui wants the device, it's your job not to check the underlying platform. It's not the job of rlImGui to change event states in the underlying platform. ImGui does not assume that the underlying platform even has events.
This is a user error issue not a missing feature in raylib.

rlImgui doesn't need to read that flag. that flat is for YOU and your code. rlImgui sends data to Imgui so it knows if it needs to set that flag or not. I am very aware of how that flag works.
In that FAQ, the 'underlying application', is your code, it is not raylib and it is not rlImGui, it's you.

**Comment by JeffM2501** [comment]
That said, rlImGui should not pull key events from the buffer when WantCaptureKeyboard is false so it can let the underlying application read those events.
But for mouse, there are no events to clear.

**Comment by JeffM2501** [comment]
rlImGui no longer eats KeyPress events, it computes them from polls, and it only eats character events if ImGui wants keyboard input.
This should allow the underlying app to get access to the events needed after checking the input capture flags in the IO structure.

**Comment by albertvaka** [comment]
This can indeed be implemented on the user app's code, it's just a bit more convoluted: either you check before each `IsMouseButtonPressed()` or you wrap that method and update all usages.

IMO it would be a quality of life improvement to have it in the application side, and also a smaller change. For an example, [here's how I implemented it](https://github.com/albertvaka/nbgf/blob/7fd094d/engine/main.cpp#L200-L213) on top of SDL with just two `if`s in total.

It might not be as simple in Raylib and maybe not worth doing it for that reason, but I do think it would make people's life easier if it was there.

**Comment by JeffM2501** [comment]
I don't think that is the correct fix for the level that rlImGui is designed to be at.
1) That change assumes the underlying app doesn't still need to poll the input data, it may for various reasons, like drawing your own cursor. I don't want rlImGui to assume anything.
2) I don't think that will work for pure keyboard navigation, once you lost capture, there would be no way to get back to it.
3) I need raylib's API to send the input TO ImGui, if I tell raylib to stop processing input I'll loose the ability to get that input next frame.
4) The field in the IO structure is from the last ImGUi frame, events this frame may change it. I don't think your fix is as robust as you thing, it may work for simple mouse based navigation.
5) I have worked with the author of ImGui in the past and I don't think the intnent of the flag is to change the underlying platform state, just tell the calling application if it should query the platform or not.
6) I think the 'convenience' of this would cause more confusion than it solves. It's your job to manage input in your app, not raylib's or ImGui. Nobody has ever asked for this in the past. In fact the problems I do get are the opposite of your request, that we are not exposing enough of the ImGui input events back to the calling app.

I have made some changes to minimize how much of the raylib events that rlImGui eats so that the app can have as much data as possible for whatever reason it needs. I don't want to assume that I know how people are using rlImGui, and it should be up to them how they handle input and if they want to use the consume flags.
